To: vim_dev@googlegroups.com
Subject: Patch 7.3.329
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.329
Problem:    When skipping over code from ":for" to ":endfor" get an error for
	    calling a dict function. (Yasuhiro Matsumoto)
Solution:   Ignore errors when skipping over :call command.
Files:	    src/ex_docmd.c, src/eval.c


*** ../vim-7.3.328/src/ex_docmd.c	2011-09-21 19:10:42.000000000 +0200
--- src/ex_docmd.c	2011-09-30 17:59:51.000000000 +0200
***************
*** 371,381 ****
  # define ex_endif		ex_ni
  # define ex_else		ex_ni
  # define ex_while		ex_ni
- # define ex_for			ex_ni
  # define ex_continue		ex_ni
  # define ex_break		ex_ni
  # define ex_endwhile		ex_ni
- # define ex_endfor		ex_ni
  # define ex_throw		ex_ni
  # define ex_try			ex_ni
  # define ex_catch		ex_ni
--- 371,379 ----
*** ../vim-7.3.328/src/eval.c	2011-09-30 16:39:41.000000000 +0200
--- src/eval.c	2011-09-30 18:29:58.000000000 +0200
***************
*** 3377,3383 ****
  	/* trans_function_name() doesn't work well when skipping, use eval0()
  	 * instead to skip to any following command, e.g. for:
  	 *   :if 0 | call dict.foo().bar() | endif  */
! 	eval0(eap->arg, &rettv, &eap->nextcmd, FALSE);
  	return;
      }
  
--- 3377,3386 ----
  	/* trans_function_name() doesn't work well when skipping, use eval0()
  	 * instead to skip to any following command, e.g. for:
  	 *   :if 0 | call dict.foo().bar() | endif  */
! 	++emsg_skip;
! 	if (eval0(eap->arg, &rettv, &eap->nextcmd, FALSE) != FAIL)
! 	    clear_tv(&rettv);
! 	--emsg_skip;
  	return;
      }
  
*** ../vim-7.3.328/src/version.c	2011-09-30 17:46:14.000000000 +0200
--- src/version.c	2011-09-30 18:32:27.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     329,
  /**/

-- 
Have you heard about the new Beowulf cluster? It's so fast, it executes
an infinite loop in 6 seconds.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
